### PR TITLE
Refactor batch start to use API

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

- `tctl batch start` now uses official API
- added unit test

## Why?

stabilizing by using the official API

## Checklist

1. Closes 

2. How was this tested:

manual + added unit test

3. Any docs updates needed?

### CODE REVIEW

File: cli/batch.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func newBatchCommands() []*cli.Command {
 				&cli.StringFlag{
 					Name:     FlagListQuery,
```
Comment (yux0) [inline cli/batch.go:68]: Not used only in list. nit: FlagQuery?

File: cli/batch_commands.go [hunk 2]
```diff
@@ -165,47 +151,63 @@ func StartBatchJob(c *cli.Context) error {
...
 	}
-	wf, err := client.ExecuteWorkflow(tcCtx, options, batcher.BatchWFTypeName, params)
-	if err != nil {
-		return fmt.Errorf("failed to start batch job: %w", err)
+
+	switch batchType {
+	case batcher.BatchTypeSignal:
+		sigName := c.String(FlagSignalName)
+
+		if sigName == "" || input == "" {
```
Comment (yux0) [inline cli/batch_commands.go:174]: we should allow empty input?

File: cli/batch_commands.go [hunk 3]
```diff
@@ -165,47 +151,63 @@ func StartBatchJob(c *cli.Context) error {
...
+			SignalOperation: &batch.BatchOperationSignal{
+				Signal:   sigName,
+				Input:    payloads,
+				Identity: operator,
+			},
+		}
+	case batcher.BatchTypeTerminate:
+		req.Operation = &workflowservice.StartBatchOperationRequest_TerminationOperation{
+			TerminationOperation: &batch.BatchOperationTermination{
+				Details:  payloads,
```
Comment (yux0) [inline cli/batch_commands.go:188]: No need to set the Details. I think this is similar to https://github.com/temporalio/tctl/blob/045ca90148ff9296e5df34beb13da652f31cd8b2/cli_curr/workflowCommands.go#L517

Comment (feedmeapples) [reply cli/batch_commands.go]: removed the details. Why is it in the API request?

File: cli/batch_commands.go (hunk 2 above)
Comment (feedmeapples) [reply cli/batch_commands.go]: addressed

File: cli/batch.go (hunk 1 above)
Comment (feedmeapples) [reply cli/batch.go]: good point

File: cli/batch_commands.go [hunk 4]
```diff
@@ -107,44 +107,30 @@ func StartBatchJob(c *cli.Context) error {
...
+
+	fmt.Printf("This batch job will be operating on %v workflows.\n", cResp.GetCount())
 	if !c.Bool(FlagYes) {
 		reader := bufio.NewReader(os.Stdin)
 		for {
 			fmt.Print("Please confirm[Yes/No]:")
 			text, err := reader.ReadString('\n')
 			if err != nil {
-				return fmt.Errorf("failed to get confirmation to start a batch job: %w", err)
+				return fmt.Errorf("unable to get confirmation to start a batch job: %s", err)
```
Comment (alexshtin) [inline cli/batch_commands.go:133]: Please use `promptYes` instead.

File: cli/batch_commands.go [hunk 5]
```diff
@@ -107,44 +107,30 @@ func StartBatchJob(c *cli.Context) error {
...
 	tcCtx, cancel := newContext(c)
 	defer cancel()
-	resp, err := client.CountWorkflow(tcCtx, &workflowservice.CountWorkflowExecutionsRequest{
+	cResp, err := sdk.CountWorkflow(tcCtx, &workflowservice.CountWorkflowExecutionsRequest{
 		Namespace: namespace,
 		Query:     query,
 	})
 	if err != nil {
-		return fmt.Errorf("failed to count impacted workflows: %w", err)
+		return fmt.Errorf("unable to count impacted workflows: %s", err)
```
Comment (alexshtin) [inline cli/batch_commands.go:123]:
```suggestion
		return fmt.Errorf("unable to count impacted workflows: %v", err)
```

File: cli/batch_commands.go [hunk 6]
```diff
@@ -30,16 +30,16 @@ import (
 	"os"
 	"strings"
 
+	"github.com/google/uuid"
```
Comment (alexshtin) [inline cli/batch_commands.go:33]: We use `"github.com/pborman/uuid"` everywhere else.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**